ninshiki_yolo/detector.py

```diff
@@ -87,22 +97,7 @@ def detection(self, image: np.ndarray) -> np.ndarray:
         image = self.postprocess(outs, image, classes)
         return image
 
-    def draw_detection_result(self, img: np.ndarray, label: str, x0: int, y0: int,
-                              xt: int, yt: int, color: tuple = (255, 127, 0),
-                              text_color: tuple = (255, 255, 255)) -> np.ndarray:
-
-        y0, yt = max(y0 - 15, 0), min(yt + 15, img.shape[0])
-        x0, xt = max(x0 - 15, 0), min(xt + 15, img.shape[1])
-
-        (w, h), baseline = cv2.getTextSize(label, cv2.FONT_HERSHEY_SIMPLEX, 0.4, 1)
-        cv2.rectangle(img, (x0, y0 + baseline), (max(xt, x0 + w), yt), color, 2)
-        cv2.rectangle(img, (x0, y0 - h), (x0 + w, y0 + baseline), color, -1)
-        cv2.putText(img, label, (x0, y0), cv2.FONT_HERSHEY_SIMPLEX, 0.4,
-                    text_color, 1, cv2.LINE_AA)
-
-        return img
-
-    def postprocess(self, outs: np.array, frame: np.array, classes: int,
+    def postprocess(self, outs: np.array, frame: np.array, classes: list,
```

Comment:
I think it's not quite right to name this function as `postprocess()`, as this function also call `add_detected_object()` in this function which should be handle by detection process, not image post process (debug result draw).

I suggest to change the process as follows:
```py
if (received_frame.size != 0):
    detection_objects = self.detection(received_frame)  # already in msg, not yolo results nor cv mat
    self.detected_object_publisher.publish(self.detection_result)
   
    # the if is optional, so later it would be possible to disable/enable the debug view
    # also, the cv2.waitKey() won't delay the detection process in non debug view mode
    if self.enable_view_or_something:
        postprocessed_frame = self.postprocess(received_frame, detection_objects)
        cv2.imshow(self.image_subscription.topic_name, postprocessed_frame)
        cv2.waitKey(1)
```